odoo/tools/template_inheritance.py

```diff
@@ -205,23 +205,94 @@ def extract(spec):
+                                    (repr(child.get('separator')), repr(attribute))
+                                )
+                            separator = child.attrib.pop('separator').strip()
+
+                            if child.get('remove'):
+                                remove = child.attrib.pop('remove')
+                                if remove == value:
+                                    value = ''
+                                elif remove in value:
+                                    is_constant = remove in ('True', 'False', '1', '0')
+                                    rf_remove = rf'(( +\()*)( *{re.escape(remove)} *)((\) +)*)' if is_constant else rf'(( +\()+)( *{re.escape(remove)} *)((\) +)+)'
+                                    while value:
+                                        part = re.search(rf_remove, value)
+                                        if not part:
+                                            break
+                                        expression = part.group(0)
+                                        start_parentesis = part.group(1)
+                                        to_remove = part.group(3)
+                                        end_parentesis = part.group(4)
+
+                                        start_n = start_parentesis.split(' (')
+                                        end_n = end_parentesis.split(') ')
+                                        if len(start_n) > len(end_n):
+                                            expression = ' ('.join(start_n[-len(end_n):]) + to_remove + end_parentesis
+                                        elif len(start_n) < len(end_n):
+                                            expression = start_parentesis + to_remove + ') '.join(end_n[0:len(start_n)])
+
+                                        if not is_constant and ' (' not in expression:
+                                            break
+
+                                        if expression == value:
+                                            value = ''
+                                            continue
+                                        elif f' {separator}{expression}' in value:
+                                            value = value.replace(f' {separator}{expression}', ' ')
+                                        elif f'{expression}{separator} ' in value:
+                                            value = value.replace(f'{expression}{separator} ', ' ')
+                                        else:
+                                            break
+
+                                    if value == ' ':
+                                        value = ''
+
+                            if child.get('add'):
+                                add = child.attrib.pop('add')
+                                value = f"{value}{separator} ({add}) " if value else add
```

Comment:
If you follow my suggestions above, this becomes
```suggestion
                            if child.get('add'):
                                add = child.attrib.pop('add')
                                value = f"({value}) {separator} ({add})" if value else add
```
which will make the code simpler and the result more intuitive.